src/ResourceManager/ApplicationInsights/Commands.ApplicationInsights/ApplicationInsights/GetApplicationInsights.cs

```diff
@@ -0,0 +1,124 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ApplicationInsights.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ApplicationInsights
+{
+    [Cmdlet(VerbsCommon.Get, ApplicationInsightsNounStr, DefaultParameterSetName = ResourceGroupParameterSet), OutputType(typeof(PSApplicationInsightsComponent))]
+    public class GetApplicationInsightsCommand : ApplicationInsightsBaseCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = false,
+            ParameterSetName = ResourceGroupParameterSet,
+            HelpMessage = "Resource Group Name.")]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            HelpMessage = "Application Insights Resource Name.")]
+        [Alias(ApplicationInsightsComponentNameAlias, ComponentNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Application Insights Component Resource Id.")]
+        [ValidateNotNull]
+        public ResourceIdentifier ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = ComponentNameParameterSet,
+            HelpMessage = "If specified, it will get back pricing plan of the application insights component.")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "If specified, it will get back pricing plan of the application insights component.")]
```

Comment:
@frankguodongchen I would update both of the help messages for this parameter to say that it also will get the daily cap and status, not just the pricing plan.

Reply:
updated. thanks!